### ISSUE TITLE
Include global custom type converters

### ISSUE THREAD

**Issue by timfish** [pplink/class-hydrator#3]
Sorry to keep adding issues, I just thought they would be better listed separately. It'll also be easier to reference them when I get some time to do some pull requests!

Some types currently don't dehydrate very well (I'm looking at you `Date`). It would be good if users could add global type converters.

**Comment by pplink** [comment]
Yeah, that would be awesome. I'm

**Comment by taemini** [comment]
Yeah! that sounds awesome if it could dehydrate and hydrate arbitrary classes whose internal tree structure is constructed with other unknown classes.
I'm considering whole new way Hydrator.provideClasses() works. It won't require you to provide internal classes.
